oauth2/handler.go

```diff
@@ -430,9 +430,13 @@ func (h *Handler) IntrospectHandler(w http.ResponseWriter, r *http.Request, _ ht
 		TokenType:       tt,
 	}
 
-	exp := resp.GetAccessRequester().GetSession().GetExpiresAt(fosite.AccessToken)
+	exp := resp.GetAccessRequester().GetSession().GetExpiresAt(tt)
 	if exp.IsZero() {
-		exp = resp.GetAccessRequester().GetRequestedAt().Add(h.AccessTokenLifespan)
+		if tt == fosite.RefreshToken {
+			exp = resp.GetAccessRequester().GetRequestedAt().Add(h.RefreshTokenLifespan)
+		} else {
+			exp = resp.GetAccessRequester().GetRequestedAt().Add(h.AccessTokenLifespan)
+		}
 	}
```

Comment:
What about the lifetime of authorization codes and id_tokens? Perhaps fetching the expiry can be extracted to a non-exported helper somewhere within this package or file.

Reply:
Also, it is possible for refresh tokens to not expire if the value of the lifespan has been set to -1. In that case, perhaps it is best to omit the expiry field in the response payload.

@aeneasr thoughts?

Reply:
> What about the lifetime of authorization codes and id_tokens? Perhaps fetching the expiry can be extracted to a non-exported helper somewhere within this package or file.

The introspection point serves access or update tokens. Other tokens are outside the specification. 
https://tools.ietf.org/html/rfc7662#section-2.1

Reply:
Ah right! 💯

Reply:
> Also, it is possible for refresh tokens to not expire if the value of the lifespan has been set to -1. In that case, perhaps it is best to omit the expiry field in the response payload.

Damn, you know this system better by me. I'm almost 💯 that's true. In that case it should be omitted.

> The introspection point serves access or update tokens. Other tokens are outside the specification.
https://tools.ietf.org/html/rfc7662#section-2.1

Probably just a typo, introspection serves **access** and **refresh** tokens. So @aaslamin point is right!